tests/spawn-utils.c

```diff
@@ -343,3 +347,65 @@ reap_async_cmd (CmdAsyncStuff *cmd,
 
   return TRUE;
 }
+
+gchar **
+build_cmd_args (CmdArg *args)
+{
+  gsize idx;
+  g_autoptr(GPtrArray) vec = string_array_new ();
+
+  for (idx = 0;; ++idx)
+    {
+      CmdArg *arg = &args[idx];
+      g_autofree gchar* arg_str = NULL;
+
+      if (arg->flag_name == NULL &&
+          arg->value == NULL)
+        break;
+
+      if (arg->flag_name != NULL &&
+          arg->value != NULL)
+        arg_str = flag (arg->flag_name, arg->value);
+      else if (arg->flag_name != NULL)
+        arg_str = g_strdup_printf ("--%s", arg->flag_name);
+      else
+        arg_str = g_strdup (arg->value);
+
+      g_ptr_array_add (vec, g_steal_pointer (&arg_str));
+    }
+
+  g_ptr_array_add (vec, NULL);
+
+  return (gchar **)g_ptr_array_free (g_steal_pointer (&vec), FALSE);
+}
+
+gchar **
+build_cmd_env (CmdEnvVar *vars)
+{
+  gsize idx;
+  g_autoptr(GPtrArray) vec = string_array_new ();
+
+  for (idx = 0;; ++idx)
+    {
+      CmdEnvVar *var = &vars[idx];
+      g_autofree gchar* env_str = NULL;
+
+      if (var->name == NULL)
+        break;
```

Comment:
Maybe put this check in the for loop condition, e.g.:

    for (idx = 0; vars[idx]->name != NULL; ++idx)